cookbook/request-user-for-authentication/README.md

```diff
@@ -0,0 +1,96 @@
+# Requesting a user to authenticate
+
+To separate the concerns, your app should not directly deal with user authentication. Rather than that, it should have a view that blends in an authentication view from another app that deals with that.
+
+Below instruction shows how to provide an authentication notice that blends in the authentication form from the [SignIn](https://github.com/StarcounterApps/SignIn) app.
+
+When your app decides that the view cannot be presented without authentication, it redirects to its own "Unauthenticated" view. 
+For example: `/your-app/partial/unauthenticated?return_uri={?}`
+
+<div class="code-name">MainHandlers.cs</div>
+
+```cs
+if (!IsSignedIn()) {
+	return Self.GET("/your-app/partial/Unauthenticated?return_uri=" + getURI);
+}
+```
+
+
+The `getURI` is the Uri requested in the original GET request.
+
+Then define a new handler for the "Unauthenticated" partial.
+
+<div class="code-name">PartialHandlers.cs</div>
+
+```cs
+Handle.GET("/your-app/partial/Unauthenticated?return_uri={?}", (string returnUri) => 
+{
+	return new UnauthenticatedPage();
+});
+```
+
+The "unauthenticated" view model could be associated with a "Unautheticated.html" which defines a warning message about not being authenticated. The main purpose for this partial is to provide the mapping to the authentication view coming from the [SignIn](https://github.com/StarcounterApps/SignIn) app using a predifined token.
+
+<div class="code-name">Unauthenticated.json</div>
+
+```json
+{
+  "Html": "/people/viewmodels/Unauthenticated.html"
+}
+```
+
+```
+<div class="code-name">Unauthenticated.json.cs</div>
+
+```cs
+using Starcounter;
+
+namespace Your-App 
+{
+    partial class Unauthenticated : Json 
+    {
+    }
+}
+```
+
+<div class="code-name">Unauthenticated.Html</div>
+
+```html
+{
+     	<template>
+		<template is="dom-bind">
+			<div>
+	    			<iron-icon icon="icons:warning"></iron-icon>
+	    			<label>You need to be signed in to use People.</label>
+			</div>
+		</template>
+	</template>
+}
+```
```

Comment:
There are redundant brackets at lines 59 and 68. Why not paste the content of https://github.com/StarcounterApps/People/blob/1d57bc7f7aec70000fab8b05553345ef8d7c5405/src/People/wwwroot/People/viewmodels/UnauthorizedPage.html?